flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/RocksDBResourceContainerTest.java

```diff
@@ -145,12 +160,34 @@ public void testGetColumnFamilyOptionsWithSharedResources() throws Exception {
 		container.close();
 	}
 
+	@Test
+	public void testCreateSharedResourcesWithExpectedCapacity() throws Exception {
+		PowerMockito.spy(RocksDBOperationUtils.class);
+		final AtomicLong actualCacheCapacity = new AtomicLong(0L);
+		// the `createCache` wrapper is introduced due to PowerMockito cannot mock on native static method easily.
+		PowerMockito.when(RocksDBOperationUtils.createCache(anyLong(), anyDouble()))
+			.thenAnswer((Answer<LRUCache>) invocation -> {
+				Object[] arguments = invocation.getArguments();
+				actualCacheCapacity.set((long) arguments[0]);
+				return (LRUCache) invocation.callRealMethod();
+			});
+
+		long totalMemorySize = 2048L;
+		double writeBufferRatio = 0.5;
+		double highPriPoolRatio = 0.1;
+		createSharedResources(totalMemorySize, writeBufferRatio, highPriPoolRatio);
+		long expectedCacheCapacity = RocksDBOperationUtils.calculateActualCacheCapacity(totalMemorySize, writeBufferRatio);
+		assertThat(actualCacheCapacity.get(), is(expectedCacheCapacity));
```

Comment:
I suggest to either remove this test, or also add a verification of write buffer manager size.